From 1a41e383fafa113b4b5482c7ca038aa659b3a3b4 Mon Sep 17 00:00:00 2001
From: Han Wang <416810799@qq.com>
Date: Sun, 23 Dec 2018 18:45:43 +0100
Subject: [PATCH 5/8] lineage-sdk: Remove useless logic on database upgrading

 * The upgradeVersion < newVersion part is never reachable because we
   increase upgradeVersion unconditionally. Make it back to stage by
   nuking it as no one has experience issues before.

Change-Id: Icac1e18c192292ad3147fd4af4c862e93ef34ba5
---
 .../LineageDatabaseHelper.java                | 20 ++-----------------
 1 file changed, 2 insertions(+), 18 deletions(-)

diff --git a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
index fc8e2ce..c94a995 100644
--- a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
+++ b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
@@ -206,13 +206,11 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
             db.beginTransaction();
             try {
                 loadSettings(db);
-
                 db.setTransactionSuccessful();
-
-                upgradeVersion = 2;
             } finally {
                 db.endTransaction();
             }
+            upgradeVersion = 2;
         }
 
         if (upgradeVersion < 3) {
@@ -399,21 +397,6 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
             upgradeVersion = 11;
         }
         // *** Remember to update DATABASE_VERSION above!
-
-        if (upgradeVersion < newVersion) {
-            Log.w(TAG, "Got stuck trying to upgrade db. Old version: " + oldVersion
-                    + ", version stuck at: " +  upgradeVersion + ", new version: "
-                            + newVersion + ". Must wipe the lineage settings provider.");
-
-            dropDbTable(db, LineageTableNames.TABLE_SYSTEM);
-            dropDbTable(db, LineageTableNames.TABLE_SECURE);
-
-            if (mUserHandle == UserHandle.USER_OWNER) {
-                dropDbTable(db, LineageTableNames.TABLE_GLOBAL);
-            }
-
-            onCreate(db);
-        }
     }
 
     @Override
@@ -679,3 +662,4 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
         stmt.execute();
     }
 }
+
-- 
2.17.1

